backend/api/viewsets/Notification.py

```diff
@@ -0,0 +1,108 @@
+import base64
+import random
+import pika
+import json
+from django.http import HttpResponse
+
+from rest_framework import viewsets, serializers, mixins, status
+from rest_framework.decorators import list_route
+from rest_framework.renderers import JSONRenderer
+from rest_framework.response import Response
+
+from api.models.NotificationMessage import NotificationMessage
+from api.notifications.notifications import AMQPNotificationService, EffectiveSubscriptionSerializer, \
+    EffectiveSubscriptionUpdateSerializer
+from api.permissions.Notifications import NotificationPermissions
+from api.serializers.Notifications import NotificationMessageSerializer
+from auditable.views import AuditableMixin
+
+
+class NotificationToken(object):
+
+    @staticmethod
+    def __generate_token():
+        t = bytearray
+        rand = random.SystemRandom()
+
+        t = bytes([rand.getrandbits(8) for x in range(48)])
+
+        return base64.encodebytes(t).decode('utf-8')
+
+    def __init__(self, token=None):
+        self.token = token or NotificationToken.__generate_token()
+
+
+class NotificationTokenSerializer(serializers.Serializer):
+    token = serializers.CharField()
+
+
+class NotificationViewSet(AuditableMixin,
+                          mixins.ListModelMixin,
+                          mixins.UpdateModelMixin,
+                          mixins.RetrieveModelMixin,
+                          viewsets.GenericViewSet):
+
+    permission_classes = (NotificationPermissions,)
+    http_method_names = ['get', 'put', 'post']
+    #serializer_class = NotificationMessageSerializer
+    serializer_classes = {
+        'update_subscription': EffectiveSubscriptionUpdateSerializer,
+        'default': NotificationMessageSerializer,
+    }
+
+    queryset = NotificationMessage.objects.all()
+    ordering = ('-id',)
+
+    def get_serializer_class(self):
+        if self.action in list(self.serializer_classes.keys()):
+            return self.serializer_classes[self.action]
+
+        return self.serializer_classes['default']
+
+    def get_queryset(self):
+        user = self.request.user
+        return NotificationMessage.objects.filter(user=user).all()
+
+    @list_route(methods=['get'])
+    def subscribe(self, request):
+        token = NotificationToken()
+        serializer = NotificationTokenSerializer(token)
+        return Response(serializer.data)
+
+    @list_route(methods=['get'])
+    def effective_subscriptions(self, request):
+        user = request.user
+        data = AMQPNotificationService.compute_effective_subscriptions(user)
+        serializer = EffectiveSubscriptionSerializer(data, many=True)
+        return Response(serializer.data)
+
+        # self.channel.basic_publish(exchange='tokens',
```

Comment:
Is this no longer needed?

Reply:
It is potentially needed for Keycloak integration. Again, I'll pull it out to a scratch file.